### PR TITLE
health record contract

### LINKED ISSUE #35: smart contract for securing patient records on blockchain
### Is your feature request related to a problem? Please describe.
The issue is surrounding the  lack of control over patient data by the patients themselves, currently all conventional hospitals/medical institutes have  complete control over the  patient data , including access and storage, at times they even sell it to make profit out of medical and personal data, to prevent such incidents and give back control to the user  over their sensitive data, we need  to move to the web3 model of smart contracts/ decentralize storage of data.

### Describe the solution you'd like
We could start with working on a solidity smart contract that could store the patient data , and then  build functions on them, to add records/delete or modify records/give access to specific contract addresses to view the records or revoke it, fetch different fields or data.

### Describe alternatives you've considered
switching to cheaper blockchains or maybe a custom blockchain built on hyperledger to meet the same goals  with cheaper trx/network fees .

### PR DESCRIPTION
**Pull request by issa-me-sush**
General health record contract,  with essential functionalities :
->create patient,doctor record.
->store the records in the form of mapping for being able to access later.
->setting permissions for viewing records based on address .
->editing /addtion of details .

fields can be further scrutinized and more functionalities can be added , security audit and gas fee reductions can be further done.

Fixes #35

### CODE REVIEW

File: contracts/Health-Rec/HealthRec.sol [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+    uint256 public permissionGrantedCount = 0;
+    
+    function Record() public {
+        owner = msg.sender;
+    }
+    
+    //Retrieve patient details from user sign up page and store the details into the blockchain
+    function setDetails(string _ic, string _name, string _phone, string _gender, string _dob, string _height, string _weight, string _houseaddr, string _bloodgroup, string _allergies, string _medication, string _emergencyName, string _emergencyContact) public {
+        require(!isPatient[msg.sender]);
+        var p = patients[msg.sender];
```
Comment (advaita-saha) [inline contracts/Health-Rec/HealthRec.sol:73]: why var?
this is not javascript

File: contracts/Health-Rec/HealthRec.sol [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+        patientList.push(msg.sender);
+        isPatient[msg.sender] = true;
+        isApproved[msg.sender][msg.sender] = true;
+        patientCount++;
+    }
+    
+    //Allows patient to edit their existing record
+    function editDetails(string _ic, string _name, string _phone, string _gender, string _dob, string _height, string _weight, string _houseaddr, string _bloodgroup, string _allergies, string _medication, string _emergencyName, string _emergencyContact) public {
+        require(isPatient[msg.sender]);
+        var p = patients[msg.sender];
```
Comment (advaita-saha) [inline contracts/Health-Rec/HealthRec.sol:100]: var not allowed in solidity ^8.0.0

**Review by advaita-saha — CHANGES_REQUESTED** [review]
Please remove `var`

**Comment by issa-me-sush** [comment]
have updated the required areas and transitioned to solidity ^0.8.0

**Comment by issa-me-sush** [comment]
also had to reduce variable count,  as it was throwing "stack too deep" upon compilation

**Review by advaita-saha — APPROVED** [review]
LGTM

**Pull request merged**